core/src/impl/Kokkos_Combined_Reducer.hpp

```diff
@@ -478,6 +496,34 @@ KOKKOS_INLINE_FUNCTION constexpr auto make_wrapped_combined_functor(
 
 template <typename FunctorType>
 using functor_has_value_t = typename FunctorType::value_type;
+
+template <typename MemberType, typename BoundaryStructType, typename Functor,
+          typename ReturnType1, typename ReturnType2, typename... ReturnTypes>
+KOKKOS_INLINE_FUNCTION void parallel_reduce_combined_reducers_impl(
+    BoundaryStructType const& boundaries, Functor const& functor,
+    ReturnType1&& returnType1, ReturnType2&& returnType2,
+    ReturnTypes&&... returnTypes) noexcept {
+  using mem_space_type = DefaultExecutionSpace::memory_space;
```

Comment:
```suggestion
  using mem_space_type = typename MemberType::execution_space::memory_space;
```

Reply:
How comes no test failed?

Reply:
For the outer-level `parallel_reduce`, we decide whether we need to call deep_copy or whether we can directly write the reduced value back. AFAICT, this doesn't apply to the nested one since we know that we can directly write the reduced value.

Reply:
Precisely what @masterleinad described. `parallel_reduce` called from the host code needs to consider in which memory space the final reduced value resides before a deep_copy back to the host memory, but the team `parallel_reduce` implementation in this PR expects to be called only from the device code with one of the team policies: `TeamThreadRange`, `ThreadVectorRange` or `TeamVectorRange`.
So, as long as the combined reducers use a consistent device memory space internally, reduction on device side shouldn't be affected.